### PR TITLE
Feat/hng 102 subscription renewal reminder template

### LINKED ISSUE #102: [FEAT] EMAIL TEMPLATES -> Subscription Renewal Reminder Template
## Description

_Create a Subscription Renewal Reminder email template according to the design using React Email_

## Acceptance Criteria

- The email template must be responsive
- The email template must be dynamic
- The email template must be done according to the design
- The email tempate must be created using React Email

### Requirements	

The following props should be available for the component

1. Title
2. Name
3. Image
4. Renewal Date
5. Renewal Price
6. Renewal Period
7. Review Subscription Link
8. Renew Subscription Link
10. Any other required props

### Tests

- [ ] Ensure template is responsive
- [ ] Ensure template can receive dynamic data according to what will be displayed
- [ ] Ensure template is displayed properly according to design and requirements

###  Tasks

- [ ] Create one dynamic email template using React Email
- [ ] Add props according to the type of text and data it will display (check the design to confirm what will be displayed)
- [ ] Template must be responsive on Bigger screens, tablet screens and mobile screens
- [ ] Create a unit test to ensure template works as expected

### TimeFrame

**The deadline for submission is Saturday 11:59PM**

# LINKS

[`FIGMA LINK`](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2156-26950&m=dev).
[`REACT EMAIL DOCS`](https://react.email/docs/components/html).

# IMAGES
Subscription Renewal Reminder (3)
Subscription Renewal Reminder (2)
Subscription Renewal Reminder (1)
Subscription Renewal Reminder

### PR DESCRIPTION
**Pull request by callmestiles**
# Description
Built a Subscription Renewal Email Template with react-email

Closes #102 

# Changes proposed

## What were you told to do?

Create a Subscription Renewal Reminder email template according to the design using React Email

## What did you do?

In the email/templates folder a new email template was created for subscription renewal. This comprised of a Header and Footer component styled with tailwind and also a SubscriptionRenewalEmail component built with React-email's components and styled with tailwind with the provided Tailwind wrapper. In the test folder, a new test was added to test if the template renders content as expected. The results of the test and a screenshot of the page will be provided below.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/Videos

Figma link: https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2156-26950&m=dev
screencapture-localhost-3000-email-2024-07-19-22_50_08
screencapture-localhost-3000-email-2024-07-19-22_47_41
image
image

### CODE REVIEW

File: src/test/SubscriptionEmailTemplate.test.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { render, screen } from "@testing-library/react";
+
+import "@testing-library/jest-dom";
```
Comment (ickynavigator) [inline src/test/SubscriptionEmailTemplate.test.tsx]: this is already imported in the `setup.ts` file

File: src/test/SubscriptionEmailTemplate.test.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    companyName: "Boilerplate Inc.",
+    faqUrl: "https://example.com/faq",
+    helpUrl: "https://example.com/help",
+    unsubscribeUrl: "https://example.com/unsubscribe",
+  };
+
+  it("renders text content correctly", () => {
+    expect.hasAssertions();
+    const { container } = render(<SubscriptionRenewalEmail {...properties} />);
+    expect(container.innerHTML).toContain(properties.title);
```
Comment (ickynavigator) [inline src/test/SubscriptionEmailTemplate.test.tsx]: use getbytext/findbytext instead

File: src/email/templates/subscription-renewal-reminder/Subscription.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            <div className="px-[5.6rem]">
+              <Section>
+                <Text className="m-0 font-medium text-[#111]">Regards,</Text>
+                <Text className="mt-0 font-medium text-[#111]">
+                  {properties.companyName}
+                </Text>
+                <Text className="my-[5.6rem] text-[1.6rem] font-normal leading-[1.5]">
+                  If you have questions, please visit our{" "}
+                  <Link href={properties.faqUrl}>
+                    <strong className="font-semibold text-[#F97316]">
```
Comment (ickynavigator) [inline src/email/templates/subscription-renewal-reminder/Subscription.tsx]: don't use custom colors in the app (reference the tailwind config/globals.css

**Review by ickynavigator — CHANGES_REQUESTED** [review]

**Comment by ickynavigator** [comment]
did you check if your email renders on react email btw?

File: src/email/templates/subscription-renewal-reminder/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        </Link>
+        <Link href="/">
+          <Image
+            className="footer__icon"
+            src="/icon-linkedin.svg"
+            alt="Linkedin icon"
+            width={24}
+            height={24}
+          />
+        </Link>
```
Comment (abiona01) [inline src/email/templates/subscription-renewal-reminder/Footer.tsx]: Just a tiny bit of nitpick, you could create an array of objects for the social media accounts then map through here. Watchu think?

File: src/test/SubscriptionEmailTemplate.test.tsx (hunk 1 above)
Comment (callmestiles) [reply src/test/SubscriptionEmailTemplate.test.tsx]: fixed

File: src/test/SubscriptionEmailTemplate.test.tsx (hunk 2 above)
Comment (callmestiles) [reply src/test/SubscriptionEmailTemplate.test.tsx]: fixed

File: src/email/templates/subscription-renewal-reminder/Subscription.tsx (hunk 3 above)
Comment (callmestiles) [reply src/email/templates/subscription-renewal-reminder/Subscription.tsx]: fixed

**Comment by callmestiles** [comment]
> did you check if your email renders on react email btw?

> did you check if your email renders on react email btw?

yes i did

File: src/email/templates/subscription-renewal-reminder/Footer.tsx (hunk 4 above)
Comment (callmestiles) [reply src/email/templates/subscription-renewal-reminder/Footer.tsx]: implemented, thank you

**Pull request closed without merging**